src/metal.cs

```diff
@@ -3614,167 +4200,552 @@ partial interface MTLRenderCommandEncoder : MTLCommandEncoder {
+#if XAMCORE_5_0
+		[Abstract]
+#endif
+		[Export ("setObjectSamplerState:lodMinClamp:lodMaxClamp:atIndex:")]
+		void SetObjectSamplerState ([NullAllowed] IMTLSamplerState sampler, float lodMinClamp, float lodMaxClamp, nuint index);
+
+		[Mac (13, 0), iOS (16, 0), TV (16, 0), MacCatalyst (16, 0)]
+#if XAMCORE_5_0
+		[Abstract]
+#endif
+		[Export ("setObjectSamplerStates:lodMinClamps:lodMaxClamps:withRange:")]
+		void SetObjectSamplerStates (IntPtr samplers, IntPtr lodMinClamps, IntPtr lodMaxClamps, NSRange range);
+
+		[Mac (13, 0), iOS (16, 0), TV (16, 0), MacCatalyst (16, 0)]
+#if XAMCORE_5_0
+		[Abstract]
+#endif
+		[Export ("setObjectSamplerStates:withRange:")]
+		void SetObjectSamplerStates (IntPtr samplers, NSRange range);
+
+		[Mac (13, 0), iOS (16, 0), TV (16, 0), MacCatalyst (16, 0)]
+#if XAMCORE_5_0
+		[Abstract]
+#endif
+		[Export ("setObjectTexture:atIndex:")]
+		void SetObjectTexture ([NullAllowed] IMTLTexture texture, nuint index);
+
+		[Mac (13, 0), iOS (16, 0), TV (16, 0), MacCatalyst (16, 0)]
+#if XAMCORE_5_0
+		[Abstract]
+#endif
+		[Export ("setObjectTextures:withRange:")]
+		void SetObjectTextures (IntPtr textures, NSRange range);
+
+		[Mac (13, 0), iOS (16, 0), TV (16, 0), MacCatalyst (16, 0)]
+#if XAMCORE_5_0
+		[Abstract]
+#endif
+		[Export ("setObjectThreadgroupMemoryLength:atIndex:")]
+		void SetObjectThreadgroupMemoryLength (nuint length, nuint index);
+	}
+
+	[MacCatalyst (13, 1)]
+	[BaseType (typeof (NSObject))]
+	interface MTLRenderPipelineColorAttachmentDescriptor : NSCopying {
+
+		[Export ("pixelFormat", ArgumentSemantic.Assign)]
+		MTLPixelFormat PixelFormat { get; set; }
+
+		[Export ("blendingEnabled")]
+		bool BlendingEnabled { [Bind ("isBlendingEnabled")] get; set; }
+
+		[Export ("sourceRGBBlendFactor", ArgumentSemantic.Assign)]
+		MTLBlendFactor SourceRgbBlendFactor { get; set; }
+
+		[Export ("destinationRGBBlendFactor", ArgumentSemantic.Assign)]
+		MTLBlendFactor DestinationRgbBlendFactor { get; set; }
+
+		[Export ("rgbBlendOperation", ArgumentSemantic.Assign)]
+		MTLBlendOperation RgbBlendOperation { get; set; }
+
+		[Export ("sourceAlphaBlendFactor", ArgumentSemantic.Assign)]
+		MTLBlendFactor SourceAlphaBlendFactor { get; set; }
+
+		[Export ("destinationAlphaBlendFactor", ArgumentSemantic.Assign)]
+		MTLBlendFactor DestinationAlphaBlendFactor { get; set; }
+
+		[Export ("alphaBlendOperation", ArgumentSemantic.Assign)]
+		MTLBlendOperation AlphaBlendOperation { get; set; }
+
+		[Export ("writeMask", ArgumentSemantic.Assign)]
+		MTLColorWriteMask WriteMask { get; set; }
+	}
+
+	[MacCatalyst (13, 1)]
+	[BaseType (typeof (NSObject))]
+	interface MTLRenderPipelineReflection {
+		[Deprecated (PlatformName.MacOSX, 13, 0)]
+		[Deprecated (PlatformName.iOS, 16, 0)]
+		[Deprecated (PlatformName.TvOS, 16, 0)]
+		[Deprecated (PlatformName.MacCatalyst, 16, 0)]
+		[Export ("vertexArguments")]
+		[NullAllowed]
+#if NET
+		MTLArgument [] VertexArguments { get; }
+#else
+		NSObject [] VertexArguments { get; }
+#endif
```

Comment:
Can't we just use `MTLArgument[]` as the return type for legacy too?

Reply:
?

Reply:
This was not changed by me: https://github.com/xamarin/xamarin-macios/blob/main/src/metal.cs#L3756C41-L3756C41

I was looking and the white spaces changes are the ones making the diff hard to read. I believe this auto-format in this large files is an issue. Too late to disable it in this PR.

<img width="2178" alt="Screenshot_2023-11-29_at_12 31 53" src="https://github.com/xamarin/xamarin-macios/assets/2190086/32902a97-b8d2-4253-b6ec-23abd7e16991">